internal/provider/utils/store.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package utils
+
+import (
+	"sync"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+type nodeDetails struct {
+	nodeName   string
+	externalIP string
+}
+
+type providerStore struct {
+	// nodes holds information required for updating Gateway status with the Node
+	// addresses, in case the Gateway is exposed by a Service of type NodePort.
+	nodes map[string]nodeDetails
+}
+
+var providerStoreInstance *providerStore
```

Comment:
instead of global, should be part of https://github.com/envoyproxy/gateway/blob/19eed9dc1c0d2ac49b10380f5ac91c2f58e56326/internal/provider/kubernetes/controller.go#L59